### DISCUSSION TITLE
Tut on adding more wandering bots?

### DISCUSSION THREAD

**Discussion by slattyslattslatt** [trickerer/Trinity-Bots#540] [Q&A]
**Hello guys i want to add 2000 wandering bots.**

In the config files it says:  "You must have at least this number of non-spawned bots in creature_template."
am i right that i need to add manually 2000 more entries in the creature_template table with random names but with the right unit class and scriptname?
 
I added 2000 more in the database with the help of chatgpt and when i start the worldserver it crashes or it says this: 
"Only 290 out of 310 bots of enabled classes aren't spawned. Desired amount of wandering bots (2000) cannot be created. Aborting!"

can you give me a tutorial on this how to add 2000 wandering npcs bots? i am still learning this whole sql database thing

**Comment by Russianranger** [comment]
I've already done this about 4 months ago, and have a file available on the discord server that has 11k extra bots for people to download. If you want to add more, you have to add them through 4 different tables;
1)creature_template
2)creature_eqip_template
3)creature_npcbots_appearance_template
4)creature_npcbots_extras_template

And they all have to be above ID 72k. And you have to make sure they all link up, meaning ID 72001 is a human warrior named Jim, and he has all the correct attributes and ranges in all 4 files at 72001.

I just copy/pasted the existing bots to duplicate them. To make them from scratch outside of the game it is too difficult to do on a mass scale.

Best suggestion - just download the one I already created.

**Reply by slattyslattslatt** [reply]
wow thats sounds awesome can you give me the discord or the download link please?

**Reply by Russianranger** [reply]
https://discord.gg/eJ4B23jv

Will be in Shared Content and Guides Forum channel. Bear in mind, you'll have to download all 4 files, open up heidisql or Editor of choice, and run these while selecting the acore_world database. I would not recommend anything over 5500 bots. After that slowdown and instability occurs.

**Reply by songjunst** [reply]
Kako prenesti?

**Comment by Conan1996-2** [comment]
> https://discord.gg/eJ4B23jv
> 
> Will be in Shared Content and Guides Forum channel. Bear in mind, you'll have to download all 4 files, open up heidisql or Editor of choice, and run these while selecting the acore_world database. I would not recommend anything over 5500 bots. After that slowdown and instability occurs.

This is awesome... but before I import these 4 files into the database, I do have some questions. Has anyone tried the Hire Bot (70000) with this? Do the 11k Bots show up in the list, and if so does the core crash?

**Reply by Russianranger** [reply]
I think I answered your question via discord, but for others who haven't seen that, here is the response;

Yes, I’ve tried it. Hire bot works for any bots **spawned** AFTER ID 70000, so if a bot is 69,999, they won’t show. They also have to be spawned in the world somewhere. So for example, you would spawn 100 bots across 10 capitals, then you would spawn the hire bot, and they would only display the 100 bots you spawned, while the remaining 11,180 that are not spawned would not show up.

In my experience, I have not faced any crashes on my custom compiled server. I have not tested this on any repack yet, however, so bear that in mind.

And in case you’re wondering, the bot IDS I added are from 80,000 to 91,999. So if you add these bots to the world, they should appear at the npcbot hire npc. But again, you still have to manually add these bots to the world. I believe the command is 

.npcb spawn ##### 

with the ##### being their respective 5 digit ID number. When you do that, they will show up as hireable bots through that npc. If you don’t manually add them, the npc will say there are 0 available bots to hire. This is something I didn’t realize for the longest time.

In summary - get a GM character to spawn bots wherever you want in the world, then spawn the hire NPC. The hire NPC will display only the bots you manually spawned into the world. It will *not* display bots that have not been spawned, or "wandering" bots.

**Comment by Griptor** [comment]
any way i can get a link to the file too please?

**Comment by jdbhkf** [comment]
It looks like that discord server is no longer up and I'd love to get a copy of this as well. I am trying to add more wandering bots to my server and need help. Thank you!

**Comment by Conan1996-2** [comment]
I added a link to all files that were given to me. 
There are four files to download and import, minimum as explained in the readme. There is also a hire bot sql you can use as well if you want hire bots at all cities, major towns and starter areas.

https://drive.google.com/drive/folders/1Z6Fc1qA4PxZelo5c6Lfy-Ao-A9Fmg1S-
